definitions/mongodb.rb

```diff
@@ -195,7 +195,14 @@
 
   # service
   service new_resource.name do
-    provider Chef::Provider::Service::Upstart if node['mongodb']['apt_repo'] == 'ubuntu-upstart'
+    case node['platform']
+    when 'ubuntu'
+      if node['platform_version'] == '16.04'
+        provider Chef::Provider::Service::Systemd
+      end
+    else
+      provider Chef::Provider::Service::Upstart if node['mongodb']['apt_repo'] == 'ubuntu-upstart'
```

Comment:
This resource block will be removed within the next week so its probably a moot point anyway